src/app/util/attribute-storage.cpp

```diff
@@ -980,6 +984,34 @@ bool emberAfEndpointEnableDisable(EndpointId endpoint, bool enable)
                     endpoint, cluster->clusterId,
                     (cluster->mask & CLUSTER_MASK_CLIENT ? EMBER_AF_CLIENT_CLUSTER_TICK : EMBER_AF_SERVER_CLUSTER_TICK));
             }
+
+            // Clear out any attribute access interceptors registered for this
+            // endpoint.
+            app::AttributeAccessInterceptor * prev = nullptr;
+            app::AttributeAccessInterceptor * cur  = attributeAccessInterceptors;
+            while (cur)
+            {
+                app::AttributeAccessInterceptor * next = cur->GetNext();
+                if (cur->MatchesExactly(endpoint))
+                {
+                    // Remove it from the list
+                    if (prev)
+                    {
+                        prev->SetNext(next);
+                    }
+                    else
+                    {
+                        attributeAccessInterceptors = next;
+                    }
+
+                    // Do not change prev in this case.
+                }
+                else
+                {
+                    prev = cur;
+                }
+                cur = next;
+            }
```

Comment:
If I understand correctly, the attribute interceptor will be disabled here and will be re-enable since `initializeEndpoint` ends up calling the defined `CLUSTER_MASK_INIT_FUNCTION` ?

At some point we would need to make that clearer and probably makes some helper for the code that register the interceptor, as it will likely be duplicated over the clusters.
```
bool IsRegistered() const { return sIsRegistered; }

     void RegisterIfNeeded()
     {
         if (!sIsRegistered)
         {
             registerAttributeAccessInterceptor(this);
         }
         sIsRegistered = true;
     }
```

Reply:
> If I understand correctly, the attribute interceptor will be disabled here and will be re-enable since initializeEndpoint ends up calling the defined CLUSTER_MASK_INIT_FUNCTION ?

That's the setup, yes.  For interceptors which are endpoint-specific, which mostly matters if a cluster impl is not actually being used on all endpoints for a given cluster id.

> and probably makes some helper for the code that register the interceptor, as it will likely be duplicated over the clusters.

That's a good point.  Something like EndpointIndependentAttributeInterceptor or something like that would work...

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/9694 on this followup.